### PR TITLE
ci: added github actions workflow

### PR DESCRIPTION
**Pull request by audrium**
Since building docker image takes a very long time (example: https://github.com/audrium/reana-env-root6/runs/1351920523) it was decided not to build the docker image, but use a `Dockerfile` linter for validation.

Here is an example of GH actions workflow run: https://github.com/audrium/reana-env-root6/runs/1352926699

### CODE REVIEW

**Comment by tiborsimko** [comment]
As discussed elsewhere, we can:

1. Introduce a `Dockerfile` verifier/checher/linter for MVP. This could actually be useful for other repositories as well.

2. Think about moving from Ubuntu to CentOS where ROOT packages are provided natively. This will (i) reduce docker image build times and (ii) move to HEP-standard OS to run ROOT analyses on. However, we would have to test all `reana-demo-*` examples that use ROOT, to make sure everything runs well. We can do this at a later stage.

**Comment by audrium** [comment]
Found a nice `Dockerfile` linter called [hadolint](https://github.com/hadolint/hadolint). Seems like a most widely used tool for this case. There are even dedicated wiki pages for each linting error with examples: [example here](https://github.com/hadolint/hadolint/wiki/DL3001)

**Review by tiborsimko — APPROVED** [review]
Looks good :+1: We can see later about (1) propagating this to all the repositories one day; (2) configuring nice list of to-be-ignored hadolint warnings across our repositories.

For now, this PR is good, and we can agree to configure `hadolint` checks in our favourite editors to also catch these locally...

File: Dockerfile [hunk 1]
```diff
@@ -38,6 +38,7 @@ RUN apt-get -y update && \
         python-pip \
         xlibmesa-glu-dev \
         zlib1g-dev && \
+    rm -rf /var/lib/apt/lists/* \
```
Comment (tiborsimko) [inline Dockerfile:41]: Just for the record I was rebuilding the image as per the minor `Dockerfile` change and it took 50m 26s.

Comment (tiborsimko) [reply Dockerfile:41]: ... and it needs ` && \` and the end so that the commands would run.

Comment (audrium) [reply Dockerfile]: Was that build faster or slower compared to before? For GHA it took 1h 25m  :)

Comment (tiborsimko) [reply Dockerfile]: The build time that I quoted was for my laptop... so same as before, since this PR does not really change things there.

**Pull request merged**